### ISSUE TITLE
Detect usage of `CharArray.toString` and `String(CharArray)`

### ISSUE THREAD

**Issue by yogurtearl** [detekt/detekt#6305]
Converting `CharArray` to a `String` can have surprising and inconsistent results. 

Any usage of `CharArray` where it might be converted to a `String`, but doesn't not use `.concatToString()` or `.contentToString()` is potentially a bug. 

It should detect this and recommend usage of `.concatToString()`. 

Specifically in a multi-platform context this can be a problem. 

Example: 
```kotlin
val array= "hello😅".toCharArray()

print("array : "); println(array)
println("array : " + array)
println("\"\$array\" : " + "$array")
println("array.toString() : " + array.toString())
println("array.concatToString() : " + array.concatToString())
println("array.contentToString() : " + array.contentToString())
println("String(array) : " + String(array)) // only compiles on Kotlin/JVM
```

on Kotlin/JVM this prints: 
```
array : hello😅
array : [C@7cc355be
"$array" : [C@7cc355be
array.toString() : [C@7cc355be
array.concatToString() : hello😅
array.contentToString() : [h, e, l, l, o, ?, ?]
String(array) : hello😅
```

on Kotlin/JS IR 1.9.0:
```
array : 104,101,108,108,111,55357,56837
array : 104,101,108,108,111,55357,56837
"$array" : 104,101,108,108,111,55357,56837
array.toString() : [...]
array.concatToString() : hello😅
array.contentToString() : [h, e, l, l, o, �, �]
```

on Kotlin/JS 1.9.0:
```
array : 104,101,108,108,111,55357,56837
array : [...]
"$array" : 104,101,108,108,111,55357,56837
array.toString() : 104,101,108,108,111,55357,56837
array.concatToString() : hello😅
array.contentToString() : [h, e, l, l, o, �, �]
```

on Kotlin/WASM 1.9.0:
```
array : kotlin.CharArray@1852998399
array : kotlin.CharArray@1852998399
"$array" : kotlin.CharArray@1852998399
array.toString() : kotlin.CharArray@1852998399
array.concatToString() : hello😅
array.contentToString() : [h, e, l, l, o, �, �]
```

**Comment by cortinico** [comment]
This is a great suggestion for a new rule 👍

**Comment by atulgpt** [comment]
HI @cortinico can we add these methods to `ForbiddenMethodCall`? Similar issue with different method is raised at https://github.com/detekt/detekt/issues/5977

**Comment by cortinico** [comment]
> HI @cortinico can we add these methods to `ForbiddenMethodCall`? Similar issue with different method is raised at #5977

Oh yeah that's also a good way to implement this without creating a new rule
